.github/workflows/generate-docs.yml

```
@@ -56,4 +56,106 @@ jobs:
+
+      - name: Determine deployment directory
+        id: deploy_dir
+        run: |
+          # For PRs, use consistent directory (latest commit overwrites previous)
+          SHORT_SHA=$(echo "${{ github.sha }}" | cut -c1-8)
+          echo "dir_name=${{ github.base_ref }}/pr-${{ github.event.number }}" >> $GITHUB_OUTPUT
+          echo "commit_ref=${{ github.sha }}" >> $GITHUB_OUTPUT
+          echo "short_sha=${SHORT_SHA}" >> $GITHUB_OUTPUT
+
+      - name: Deploy to internal GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
```

actionlint (1.7.7)

the runner of "peaceiris/actions-gh-pages@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)